gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/flow/FlowMongoRepository.java

```diff
@@ -35,4 +35,7 @@ public interface FlowMongoRepository extends MongoRepository<FlowMongo, String>
 
     @Query(value = "{ 'referenceId': ?0, 'referenceType': ?1}", fields = "{ _id : 1 }", delete = true)
     List<FlowMongo> deleteByReferenceIdAndReferenceType(String referenceId, String referenceTYpe);
+
+    @Query(value = "{ 'referenceType': ?0, 'referenceId': ?1 }", sort = "{referenceId: -1, order: -1}")
```

Comment:
since you sort by order maybe worth to add a new index for mongo with referenceType, referenceId and order? 
For now we have only referenceType, referenceId